### PR TITLE
Add a versions Table, define versioning scheme, and bump version to 1.0.0

### PR DESCRIPTION
**Pull request by mwachs5**
~Since 0.4.0 we added a lot of information missing from the spec so seems we should bump and re-release the doc.~ The 0.4.0 version bump post-dated adding the missing stuff. So changing this PR to  say something about the versioning scheme of this document and when to bump each version.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ author:
 title: Specification for the FIRRTL Language
 date: \today
 # Custom options added to the customized template
-version: 0.4.0
+version: 0.4.1
```
Comment (mwachs5) [inline spec.md:7]:
```suggestion
version: 1.0.0
```

File: spec.md [hunk 2]
```diff
@@ -43,6 +43,11 @@ secPrefix:
 lastDelim: ", and"
 ---
 
+# Versions
+
+- 0.4.1 Clarify the versioning scheme of this specification.
```
Comment (mwachs5) [inline spec.md:48]:
```suggestion
- 1.0.0 Clarify the versioning scheme of this specification.
```

File: spec.md [hunk 3]
```diff
@@ -2895,3 +2900,22 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
             { module | extmodule } ,
           dedent ;
 ```
+
+
+# Versioning Scheme of this Document
+
+This is the versioning scheme that applies 0.4.0 and later.
```
Comment (mwachs5) [inline spec.md:2907]:
```suggestion
This is the versioning scheme that applies 1.0.0 and later.
```

File: spec.md [hunk 4]
```diff
@@ -2895,3 +2900,22 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
...
+Specifically, 
+
+The PATCH digit is bumped upon release which only includes non-functional changes, such as grammar edits,
+further examples, and clarifications.
+
+The MINOR digit is bumped for feature additions to the spec.
+
+The MAJOR digit is bumped for backwards-incompatible changes such as features being removed from the spec or changing their interpretation.
+
+In other words, Any `.fir` file that was compliant with `x.y.z` will be compliant with `x.Y.Z`, where `Y >= y`, z and Z can be any number.
```
Comment (mwachs5) [inline spec.md:2920]:
```suggestion
In other words, Any `.fir` file that was compliant with `x.y.z` will be compliant with `x.Y.Z`, where `Y >= y`, `z` and `Z` can be any number.
```

File: spec.md [hunk 5]
```diff
@@ -2895,3 +2900,22 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
...
+The versioning scheme applies [Semantic Versioning 2.0.0](https://semver.org/#semantic-versioning-200).
+
+Specifically, 
+
+The PATCH digit is bumped upon release which only includes non-functional changes, such as grammar edits,
+further examples, and clarifications.
+
+The MINOR digit is bumped for feature additions to the spec.
+
+The MAJOR digit is bumped for backwards-incompatible changes such as features being removed from the spec or changing their interpretation.
```
Comment (mwachs5) [inline spec.md:2918]:
```suggestion
The MAJOR digit is bumped for backwards-incompatible changes such as features being removed from the spec, changing their interpretation, or new required features being added to the specification.
```

File: spec.md [hunk 6]
```diff
@@ -43,6 +43,11 @@ secPrefix:
 lastDelim: ", and"
 ---
 
+# Versions
```
Comment (mwachs5) [inline spec.md:46]:
```suggestion
# Revision History
```

File: spec.md [hunk 7]
```diff
@@ -43,6 +43,11 @@ secPrefix:
...
+
+- 1.0.0 Clarify the versioning scheme of this specification.
+- 0.4.0 Doc moved to Markdown
```
Comment (seldridge) [inline spec.md:49]: This was actually 0.3.0. Could you summarize the release notes here: https://github.com/chipsalliance/firrtl-spec/releases/tag/v0.4.0

File: spec.md [hunk 8]
```diff
@@ -2895,3 +2900,22 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
...
+
+The PATCH digit is bumped upon release which only includes non-functional changes, such as grammar edits,
+further examples, and clarifications.
+
+The MINOR digit is bumped for feature additions to the spec.
+
+The MAJOR digit is bumped for backwards-incompatible changes such as features being removed from the spec, changing their interpretation, or new required features being added to the specification.
+
+In other words, Any `.fir` file that was compliant with `x.y.z` will be compliant with `x.Y.Z`, where `Y >= y`, `z` and `Z` can be any number.
+
```
Comment (seldridge) [inline spec.md:2921]: Nit: line-wrap to 80-characters. 😉

**Review by seldridge — APPROVED** [review]

File: spec.md [hunk 9]
```diff
@@ -43,6 +43,11 @@ secPrefix:
 lastDelim: ", and"
 ---
 
+# Revision History
+
+- 1.0.0 Clarify the versioning scheme of this specification.
+- 0.4.0 Doc moved to Markdown
```
Comment (mwachs5) [inline spec.md:49]:
```suggestion
- 0.4.0
  - Add documentation for undocumented features of
the Scala-based FIRRTL Compiler (SFC) that are de facto a part of the
FIRRTL specification due to their widespread use in Chisel and the SFC:  Annotations, Targets, Asynchronous Reset,  Abstract Reset
  - There are also minor typo corrections and prose clarifications.
- 0.3.0 Doc moved to Markdown
```

File: spec.md [hunk 10]
```diff
@@ -43,6 +43,17 @@ secPrefix:
...
+# Revision History
+
+* 1.0.0 Document the versioning scheme of this specification.
+* 0.4.0
+  - Add documentation for undocumented features of the Scala-based
+    FIRRTL Compiler (SFC) that are de facto a part of the FIRRTL specification
+    due to their widespread use in Chisel and the SFC: Annotations, Targets,
+    Asynchronous Reset,  Abstract Reset
+  - Minor typo corrections and prose clarifications.
+* 0.3.0 Document moved to Markdown
```
Comment (mwachs5) [inline spec.md:55]:
```suggestion
* 0.3.1
  - Clarify analog usage in registers
  - Rework authorship as "The FIRRTL Specification Contributors"
  - Add version information as subtitle
  - Formatting fixes
* 0.3.0 Document moved to Markdown
```

File: spec.md [hunk 11]
```diff
@@ -2895,3 +2911,26 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
             { module | extmodule } ,
           dedent ;
 ```
+
+
+# Versioning Scheme of this Document
+
+This is the versioning scheme that applies 1.0.0 and later.
+
+The versioning scheme applies
```
Comment (mwachs5) [inline spec.md:2920]:
```suggestion
The versioning scheme complies with
```

File: spec.md [hunk 12]
```diff
@@ -2895,3 +2911,26 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
             { module | extmodule } ,
           dedent ;
 ```
+
+
+# Versioning Scheme of this Document
+
+This is the versioning scheme that applies 1.0.0 and later.
```
Comment (seldridge) [inline spec.md:2918]: Nit: 
```suggestion
This is the versioning scheme that applies to version 1.0.0 and later.
```

File: spec.md [hunk 13]
```diff
@@ -2895,3 +2912,26 @@ circuit = "circuit" , id , ":" , [ info ] , newline , indent ,
...
+The PATCH digit is bumped upon release which only includes non-functional changes,
+such as grammar edits, further examples, and clarifications.
+
+The MINOR digit is bumped for feature additions to the spec.
+
+The MAJOR digit is bumped for backwards-incompatible changes such as features
+being removed from the spec, changing their interpretation, or new required
+features being added to the specification.
+
+In other words, Any `.fir` file that was compliant with `x.y.z` will be compliant
```
Comment (mwachs5) [inline spec.md:2935]:
```suggestion
In other words, any `.fir` file that was compliant with `x.y.z` will be compliant
```

**Review by seldridge — APPROVED** [review]
